x-pack/elastic-agent/pkg/agent/cmd/enroll_cmd.go

```diff
@@ -401,6 +405,11 @@ func (c *enrollCmd) prepareFleetTLS() error {
 	if c.options.URL == "" {
 		return errors.New("url is required when a certificate is provided")
 	}
+
+	if c.options.FleetServer.InternalPort > 0 {
+		c.options.InternalURL = fmt.Sprintf("%s:%d", defaultFleetServerInternalHost, c.options.FleetServer.InternalPort)
```

Comment:
Can we add a long entry here which internal port is used and that this is not "officially" supported?

Reply:
i can put it there, but as soon as you have something on a machine which binds this port there needs to be official workaround and i think this is it

Reply:
Lets wait until this happens. Having the log message and user pinging us on it will also help us detect if this might become a common problem and the port we selected is not ideal.